indico/web/client/js/react/components/category_navigator/viewmodel.jsx

```diff
@@ -0,0 +1,131 @@
+// This file is part of Indico.
+// Copyright (C) 2002 - 2024 CERN
+//
+// Indico is free software; you can redistribute it and/or
+// modify it under the terms of the MIT License; see the
+// LICENSE file for more details.
+
+import PropTypes from 'prop-types';
+import React, {useEffect, useRef, useState} from 'react';
+import ReactDOM from 'react-dom';
+
+import {getCategoryData, getCategorySearchResults} from './api';
+import * as model from './model';
+
+const loadingSpinnerDelay = 1500;
+
+export function DialogViewModel({trigger, categoryId, view: View, actionView, dialogTitle}) {
+  const [state, setState] = useState(model.init());
```

Comment:
Not sure, but I've been using this pattern since before useReducer was invented. 🤔

Reply:
I'm fine with leaving it (even though it would probably not be much work to change), was just a bit surprised to see *you* kind of re-inventing the wheel! ;)